libnd4j/include/legacy/cuda/NativeOpExecutioner.cu

```diff
@@ -118,125 +111,155 @@ void NativeOpExecutioner::execPairwiseTransform(sd::LaunchContext* lc, int opNum
+    THROW_EXCEPTION(errorMessage.c_str());
+  }
 
+  if (!DataTypeUtils::isZ(zType)) {
+    std::string errorMessage;
+    errorMessage += "NativeOpExecutioner::execPairwiseIntTransform requires Z operand to have INT type";
+    errorMessage += "X type: ";
+    errorMessage += DataTypeUtils::asString(xType);
+    errorMessage += "Y type: ";
+    errorMessage += DataTypeUtils::asString(yType);
+    errorMessage += "Z type: ";
+    errorMessage += DataTypeUtils::asString(zType);
+    THROW_EXCEPTION(errorMessage.c_str());
+  }
+  if (yType != xType || zType != xType) {
+    std::string errorMessage;
+    errorMessage += "NativeOpExecutioner::execPairwiseIntTransform both operands must have same data type x type:";
+    errorMessage += DataTypeUtils::asString(xType);
+    errorMessage += " y type: ";
+    errorMessage += DataTypeUtils::asString(yType);
+    THROW_EXCEPTION(errorMessage.c_str());
+  }
   dim3 launchDims = getLaunchDims("pairwiseTransforms");
 
   BUILD_SINGLE_SELECTOR(
       xType, functions::pairwise_transforms::PairWiseIntTransform,
       ::executeCudaShaped(launchDims, stream, opNum, dX, dXShapeInfo, dY, dYShapeInfo, dZ, dZShapeInfo, extraParams),
       SD_INTEGER_TYPES)
-
-
 }
 
 ////////////////////////////////////////////////////////////////////////
-void NativeOpExecutioner::execSummaryStatsScalar(sd::LaunchContext* lc, int opNum, void const* hX,
-                                                 sd::LongType const* hXShapeInfo, void const* dX,
-                                                 sd::LongType const* dXShapeInfo, void* extraParams, void* hZ,
-                                                 sd::LongType const* hZShapeInfo, void* dZ,
-                                                 sd::LongType const* dZShapeInfo, bool biasCorrected) {
+void NativeOpExecutioner::execSummaryStatsScalar(LaunchContext* lc, int opNum, void const* hX,
+                                                 LongType const* hXShapeInfo, void const* dX,
+                                                 LongType const* dXShapeInfo, void* extraParams, void* hZ,
+                                                 LongType const* hZShapeInfo, void* dZ, LongType const* dZShapeInfo,
+                                                 bool biasCorrected) {
   auto stream = lc->getCudaStream();
   auto reductionPointer = lc->getReductionPointer();
 
-  dim3 launchDims =  getLaunchDims("summaryStats");
-
-  auto xType = sd::ArrayOptions::dataType(hXShapeInfo);
-  auto zType = sd::ArrayOptions::dataType(hZShapeInfo);
+  dim3 launchDims = getLaunchDims("summaryStats");
 
+  auto xType = ArrayOptions::dataType(hXShapeInfo);
+  auto zType = ArrayOptions::dataType(hZShapeInfo);
+  if (DataTypeUtils::isS(xType) || DataTypeUtils::isS(zType)) {
+    THROW_EXCEPTION(
+        "NativeOpExecutioner::execSummaryStatsScalar:: unable to execute on strings. Please write logic higher level "
+        "in each op for the string data type.")
```

Comment:
Quotes w/o concatenation and a missing semi colon. This appears a couple more times in this file.

Reply:
Ah great catch! Let me push up a fix for this. Thanks! I'm currently in the middle of something else but should be able to get to this really fast in a commit.